src/unitxt/inference.py

```diff
@@ -351,8 +351,8 @@ class WMLInferenceEngine(InferenceEngine, PackageRequirementsMixin):
         results = wml_inference.infer(dataset["test"])
     """
 
-    client = None
-    credentials = None
+    client: Any = None
```

Comment:
Typing is important to allow override (that is an odd python behavior), but can we use stricter types than Any?

Reply:
Initially I wanted to use types from `ibm_watsonx_ai` but it was causing errors since the package needs to be installed. And not everyone may want to install it as well as the package is incompatible with older Python versions. Having said this, I described both variables in the documentation

Reply:
Ok. Make sense.